src/middlewares/entity-audit.middleware.ts

```
@@ -0,0 +1,106 @@
+      req.preAuditEntity = entity;
+      req.entityType = entityType;
+
+      // Add event listeners for response completion
+      res.on("finish", async () => {
+        // This runs after the response has been sent
+        // Only audit successful operations (2xx status codes)
+        if (res.statusCode >= 200 && res.statusCode < 300) {
+          try {
+            // Check request method to determine operation type
+            let action: AuditLogActionsEnum;
+            let oldValues: Record<string, any> | undefined;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)